module/VuFind/src/VuFind/View/Helper/Root/MakeLink.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Make link view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+/**
+ * Make link view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class MakeLink extends \Laminas\View\Helper\AbstractHelper
+{
+    /**
+     * Render an HTML link
+     *
+     * $href will override $attrs['href']
+     * > Feel free to use like makeLink('text', 'href', $defaults);
+     *
+     * If no $href, will try to find an href in $attrs
+     * > makeLink('text', null, ['href' => '#', 'class' => 'btn-link'])
+     *
+     * If $attrs is a string, will be treated like a class
+     * > makeLink('text', $href, 'btn-link')
+     *
+     * Additional options
+     * - proxyUrl: proxy url prefix before href
+     * - escapeContent: Default true, set to false to skip escaping (like for HTML).
+     *
+     * @param string       $innerHtml Link contents (must be properly-formed HTML)
+     * @param string|array $href      Link destination (null to skip)
+     * @param array        $attrs     Link attributes (associative array)
+     * @param array        $options   Additional options
+     *
+     * @return string HTML for an anchor tag
+     */
+    public function __invoke(
+        string $innerHtml,
+        string $href = null,
+        $attrs = [],
+        $options = []
+    ) {
+        // If $attrs is not an object, interpret as class name
+        if (!is_array($attrs)) {
+            $attrs = !empty($attrs) ? ['class' => $attrs] : [];
+        }
+
+        // Merge all attributes
+        $mergedAttrs = array_merge(
```

Comment:
I wonder if it would make sense to refactor so there is a getMergedAttrs() method, so we have a more specific place to put the 'proxyUrl' option handling. This way, if we try to contribute this helper upstream to Laminas and they don't want that very specialized behavior, we can easily reimplement it in a subclass without having to rewrite this entire method.

Reply:
Not sure I quite understand

Reply:
Right now, you're doing a lot of attribute processing, including some fairly specialized logic, inside the __invoke method. I'm proposing creating a protected support method to do the attribute processing separately. I think this would make the code a little more readable and extensible. Happy to propose a refactor if you're still not sure what I mean.

I'm proposing this in part because we had talked about contributing this helper upstream to Laminas. If we do that, I suspect that some of these attribute processing rules might not be generic enough for broad use... if that's the case, if we do attribute processing in its own method, we can remove some things that others don't want and reimplement them as a VuFind-specific subclass of the Laminas helper, and we can keep things streamlined.